algo/src/main/java/com/hartwig/serve/sources/ckb/region/CkbRegion.java

```diff
@@ -0,0 +1,40 @@
+package com.hartwig.serve.sources.ckb.region;
+
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import com.hartwig.serve.ckb.datamodel.clinicaltrial.Location;
+
+import org.immutables.value.Value;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+@Value.Immutable
+@Value.Style(passAnnotations = { NotNull.class, Nullable.class })
+public abstract class CkbRegion {
+
+    @NotNull
+    public abstract String country();
+
+    @NotNull
+    public abstract Set<String> states();
+
+    public boolean includes(@NotNull Location location) {
+        return country().equals(toLowercaseNullable(location.country())) && (states().isEmpty() || states().contains(toLowercaseNullable(
+                location.state())));
+    }
+
+    @NotNull
+    public static CkbRegion of(@NotNull String country, @NotNull Set<String> states) {
+        return ImmutableCkbRegion.builder()
+                .country(country.toLowerCase())
+                .states(states.stream().map(String::toLowerCase).collect(Collectors.toSet()))
+                .build();
+    }
```

Comment:
Also minor but it's not typical to have this function. The production code generally only has one place to instantiate (where we would use the Immutable directly) and for test we'd use a test factory. If you want to leave this in place fine with me though

Reply:
Ok, this a pattern I normally use to allow clients of the immutable not to have to know they are using an immutable (ie depend only on the interface). But I think its most important to be consistent, so I'll move this elsewhere.